sympy/combinatorics/homomorphisms.py

```diff
@@ -487,21 +487,28 @@ def group_isomorphism(G, H):
 
         if not counterexample:
             # Returns a `GroupHomomorphism` object.
-            return homomorphism(G, H, G.generators, H, check=False)
+            return homomorphism(G, H, G.generators, subset, check=False)
 
     return None
 
-def is_isomorphic(G, H):
+def is_isomorphic(G, H, isomorphism=False):
     '''
     Check if the given groups are isomorphic.
 
     Arguments:
         G (a finite `FpGroup` or a `PermutationGroup`) -- First group
         H (a finite `FpGroup` or a `PermutationGroup`) -- Second group
+        isomorphism(boolean) -- This computes the isomorphism when set to True
 
     Returns:
-    `True` if the groups are isomorphic
-    `False` if the groups are not isomorphic
+    if isomorphism is set to `False` --
+        `True` if the groups are isomorphic
+        `False` if the groups are not isomorphic
+    if isomorphism is set to `True` --
+        Returns a tuple (boolean, `GroupHomomorphism` object).
 
     '''
-    return bool(group_isomorphism(G, H))
+    T = group_isomorphism(G, H)
+    if isomorphism:
+        return (T.is_isomorphism() , T)
```

Comment:
There is not method `is_isomorphism` anymore. Beside, running `T.is_isomorphic` would just recompute `T` again - we don't want this. You should just check here if `T` is not `None` to decide whether to return `True` or `False`. Also, you are currently not returning anything if `isomorphism=False`.

Reply:
> There is not method is_isomorphism anymore. 

This is the method of the `GroupIsomorphism` object. I could use bool(T) but, I thought that checking the isomorphism using the existing also would be good.

Reply:
Ah, I see. Still, I think this sort of thing would be very good to use in tests (perhaps, you should add this somewhere in the current tests?), but not in the actual code, because `is_isomorphism` does further computations to check that `T` is actually an isomorphism - but you already know that it is (provided your algorithm is correct, of course).

Here's a thought though: `is_isomorphism` will compute the kernel of `T` to see if if `T` is injective - but you've constructed it that way, so you could say `T._kernel = PermutationGroup()` at the end of `group_isomorphism`. This way, if a user does call `is_isomorphism` on `T`, it will take less time.